tests/test_2023_10_03.py

```diff
@@ -0,0 +1,113 @@
+"""
+Tests for voting 03/10/2023.
+"""
+from brownie import ZERO_ADDRESS, convert, reverts
+from scripts.vote_2023_10_03 import start_vote
+
+from utils.test.tx_tracing_helpers import (
+    count_vote_items_by_events,
+    display_voting_events,
+    group_voting_events,
+)
+
+from utils.config import (
+    contracts,
+    LDO_HOLDER_ADDRESS_FOR_TESTS,
+    AGENT
+)
+
+from utils.test.event_validators.anchor import (
+    validate_anchor_vault_implementation_upgrade_events,
+    validate_anchor_vault_version_upgrade_events
+)
+
+ANCHOR_OLD_IMPL_ADDRESS = "0x07BE9BB2B1789b8F5B2f9345F18378A8B036A171"
+ANCHOR_NEW_IMPL_ADDRESS = "#TBA"
+EMERGENCY_ADMIN = "0x3cd9F71F80AB08ea5a7Dca348B5e94BC595f26A0"
+TERRA_ADDRESS = "0xabcdefabcdefabcdefabcdefabcdefabcdefabcdefabcdefabcdefabcdefabcd"
+REFUND_BETH_AMOUNT = 4449999990000000000 + 439111118580000000000
+ANCHOR_NEW_VERSION = 4
+
+def test_vote(helpers, accounts, vote_ids_from_env, interface, ldo_holder, stranger, bypass_events_decoding):
+
+    max_uint256 = convert.to_uint(
+        "0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff"
+    )
+
+    proxy = contracts.anchor_vault_proxy
+    vault = contracts.anchor_vault
+
+    # check that implementation is petrified
+    anchor_impl = interface.AnchorVault(ANCHOR_NEW_IMPL_ADDRESS)
+    assert anchor_impl.version() == max_uint256
+
+    address_implementation_before = proxy.implementation()
+    assert address_implementation_before == ANCHOR_OLD_IMPL_ADDRESS, "Old address is incorrect"
+
+    assert vault.version() == 3
+    assert vault.emergency_admin() == EMERGENCY_ADMIN
+
+    admin_before = vault.admin()
+    beth_token_before = vault.beth_token()
+    steth_token_before = vault.steth_token()
+    operations_allowed_before = vault.operations_allowed()
+    total_beth_refunded_before = vault.total_beth_refunded()
```

Comment:
+ let's check stETH balance of the contract AND bETH total supply before and after the vote

Reply:
done